### PR TITLE
[REVIEW] Added hs.fs.readTextFileContents(path)

### LINKED ISSUE #1452: Reading contents of a UTF-16 Little-Endian Encoded Text File?
I have a UTF-16 Little-Endian Encoded Text File that I'd like to read via Lua code. Normally I'd just use `io.lines(file)` however, that just gives me garbage.

Any ideas how I can achieve this? @asmagill or @cmsj ?

### PR DESCRIPTION
**Pull request by latenitefilms**
- Closes #1452

### CODE REVIEW

**Comment by latenitefilms** [comment]
@asmagill & @cmsj - I believe this might solve #1452.

Attached are some test files which seem to all work correctly:

[lorem_utf8.txt](https://github.com/Hammerspoon/hammerspoon/files/3744325/lorem_utf8.txt)
[lorem_utf16be.txt](https://github.com/Hammerspoon/hammerspoon/files/3744326/lorem_utf16be.txt)
[lorem_utf16le.txt](https://github.com/Hammerspoon/hammerspoon/files/3744327/lorem_utf16le.txt)
[utf8.txt](https://github.com/Hammerspoon/hammerspoon/files/3744328/utf8.txt)
[utf16be.txt](https://github.com/Hammerspoon/hammerspoon/files/3744329/utf16be.txt)
[utf16le.txt](https://github.com/Hammerspoon/hammerspoon/files/3744330/utf16le.txt)

Any questions let me know!

File: extensions/fs/internal.m [hunk 1]
```diff
@@ -1090,6 +1090,41 @@ static int fs_pathFromBookmark(lua_State *L) {
...
+///  * An error message if an error occurs.
+static int fs_readFileContents(lua_State *L) {
+    LuaSkin *skin = [LuaSkin shared] ;
+    [skin checkArgs:LS_TSTRING, LS_TBREAK] ;
+    
+    NSString *filePath = [skin toNSObjectAtIndex:1];
+    NSURL *url = [NSURL fileURLWithPath:filePath];
+    
+    NSError *error = nil;
+    NSStringEncoding encoding;
```
Comment (randomeizer) [inline extensions/fs/internal.m]: What is the encoding variable used for?

File: extensions/fs/internal.m [hunk 2]
```diff
@@ -1090,6 +1090,41 @@ static int fs_pathFromBookmark(lua_State *L) {
...
+/// Function
+/// Reads the contents of a file at a given path.
+///
+/// Parameters:
+///  * path - The path to the file you want to read.
+///
+/// Returns:
+///  * A string containing the contents of the file or `nil` if an error occurs.
+///  * An error message if an error occurs.
+static int fs_readFileContents(lua_State *L) {
```
Comment (randomeizer) [inline extensions/fs/internal.m]: In general, I'm not sure I like having a general "read file contents" function, given that files can contain anything, not just string values. Lua uses strings as general-purpose bit arrays as well, but this kind of implies a text value.

File: extensions/fs/internal.m (hunk 1 above)
Comment (latenitefilms) [reply extensions/fs/internal.m]: Nothing - it's just used to store the `NSStringEncoding` type when the `NSString` is initialised. I guess we could return it, if that's useful?

File: extensions/fs/internal.m (hunk 2 above)
Comment (latenitefilms) [reply extensions/fs/internal.m]: Good point - I've updated the function name.

**Comment by latenitefilms** [comment]
@randomeizer - Something like this?

```objc
/// hs.utf8.byteStringToEncodedString(value[, encodingType]) -> string
/// Function
/// Converts a byte string to an encoded string.
///
/// Parameters:
///  * value - The byte string to convert.
/// * encodingType - An optional string containing the supported macOS encoding type. Defaults to "NSUTF8StringEncoding".
///
/// Returns:
///  * A string
///
/// Notes:
///  * Supported `encodingType`'s:
///    * "NSASCIIStringEncoding"
///    * "NSNEXTSTEPStringEncoding"
///    * "NSJapaneseEUCStringEncoding"
///    * "NSUTF8StringEncoding"
///    * "NSISOLatin1StringEncoding"
///    * "NSSymbolStringEncoding"
///    * "NSNonLossyASCIIStringEncoding"
///    * "NSShiftJISStringEncoding"
///    * "NSISOLatin2StringEncoding"
///    * "NSUnicodeStringEncoding"
///    * "NSWindowsCP1251StringEncoding"
///    * "NSWindowsCP1252StringEncoding"
///    * "NSWindowsCP1253StringEncoding"
///    * "NSWindowsCP1254StringEncoding"
///    * "NSWindowsCP1250StringEncoding"
///    * "NSISO2022JPStringEncoding"
///    * "NSMacOSRomanStringEncoding"
///    * "NSUTF16StringEncoding"
///    * "NSUTF16BigEndianStringEncoding"
///    * "NSUTF16LittleEndianStringEncoding"
///    * "NSUTF32StringEncoding"
///    * "NSUTF32BigEndianStringEncoding"
///    * "NSUTF32LittleEndianStringEncoding"
///    * "NSProprietaryStringEncoding"
```

**Comment by latenitefilms** [comment]
@cmsj & @asmagill - Will leave you two to decide whether or not this is worth merging. No worries at all if you just want to close and ignore.

**Comment by asmagill** [comment]
If `hs.text` lands, is this still necessary?

**Comment by latenitefilms** [comment]
I don't think so. Although it would be nice to have an easier/quicker way of reading a file? `hs.fs.read()`?

**Comment by latenitefilms** [comment]
I don't think we need this now we have `hs.text.readFile()`, so closing.

**Pull request closed without merging**